crates/router/src/connector/payone.rs

```diff
@@ -47,25 +89,44 @@ impl
         types::PaymentsResponseData,
     > for Payone
 {
-    // Not Implemented (R)
 }
 
 impl<Flow, Request, Response> ConnectorCommonExt<Flow, Request, Response> for Payone
 where
     Self: ConnectorIntegration<Flow, Request, Response>,
 {
+    #[cfg(feature = "payouts")]
     fn build_headers(
         &self,
         req: &types::RouterData<Flow, Request, Response>,
-        _connectors: &settings::Connectors,
+        connectors: &settings::Connectors,
     ) -> CustomResult<Vec<(String, request::Maskable<String>)>, errors::ConnectorError> {
-        let mut header = vec![(
-            headers::CONTENT_TYPE.to_string(),
-            self.get_content_type().to_string().into(),
-        )];
-        let mut api_key = self.get_auth_header(&req.connector_auth_type)?;
-        header.append(&mut api_key);
-        Ok(header)
+        let auth = payone::PayoneAuthType::try_from(&req.connector_auth_type)?;
+        let http_method = Self.get_http_method().to_string();
+        let content_type = Self::get_content_type(self);
+        let base_url = self.base_url(connectors);
+        let url = Self::get_url(self, req, connectors)?;
+        let date_header = get_formatted_date_time(
+            "[weekday repr:short], [day] [month repr:short] [year] [hour]:[minute]:[second] GMT",
+        )?;
+        let path: String = url.replace(base_url, "/");
+
+        let authorization_header: String = self.generate_signature(
+            auth,
+            http_method,
+            path,
+            content_type.to_string(),
+            date_header.clone(),
+        )?;
+        let headers = vec![
+            (headers::DATE.to_string(), date_header.to_string().into()),
```

Comment:
Do we need to explicitly add the `Date` header or is it automatically added by `reqwest`.